src/App.tsx

```diff
@@ -1,18 +1,56 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
-import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { getNumbers, getNumbersForCurrentPage } from './utils';
+import { Pagination } from './components/Pagination/Pagination';
+import { ItemsPerPageType } from './types';
+
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage]
+    = useState<number>(ItemsPerPageType.Five);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  const itemsQuantity = items.length;
+
+  const itemsToRender = getNumbersForCurrentPage(
+    itemsQuantity,
+    currentPage,
+    itemsPerPage,
+  );
+
+  const handleCurrentPage = (event: React.MouseEvent<HTMLAnchorElement>) => {
+    const currentAnchor = event.currentTarget.href.split('#')[1];
+
+    if (currentAnchor === 'next') {
+      setCurrentPage(currentPage + 1);
+
+      return;
+    }
+
+    if (currentAnchor === 'prev') {
+      setCurrentPage(currentPage - 1);
+
+      return;
+    }
+
+    setCurrentPage(Number(currentAnchor));
+  };
+
+  const handleSelectItemsPerPage
+  = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setItemsPerPage(+event.target.value);
+    setCurrentPage(1);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${itemsToRender[0]} - ${itemsToRender[itemsToRender.length - 1]} of ${itemsQuantity})`}
```

Comment:
It is better to make calculations in separate constants and use the ready result.